src/bootstrap/src/core/build_steps/doc.rs

```diff
@@ -1292,8 +1306,19 @@ impl Step for Reference {
     }
 
     fn make_run(run: RunConfig<'_>) {
+        // Bump the stage to 2, because the reference requires an in-tree compiler.
+        // At the same time, since this step is enabled by default, we don't want `x doc` to fail
+        // in stage 1.
+        // FIXME: create a shared method on builder for auto-bumping, and print some warning when
+        // it happens.
```

Comment:
Discussion: even though a shared helper isn't a thing yet, should we already emit a warning *now* as an interim thing?

Reply:
Okay, sounds reasonable.